src/test/cli/crushtool/choose-args.crush

```diff
@@ -0,0 +1,120 @@
+# begin crush map
+
+# devices
+device 0 device0
+device 1 device1
+device 2 device2
+
+# types
+type 0 device
+type 1 host
+type 2 rack
+type 3 root
+
+# buckets
+host host0 {
+	id -1		# do not change unnecessarily
+	# weight 1.000
+	alg straw
+	hash 0	# rjenkins1
+	item device0 weight 1.000
+}
+host host1 {
+	id -2		# do not change unnecessarily
+	# weight 1.000
+	alg straw
+	hash 0	# rjenkins1
+	item device1 weight 1.000
+}
+host host2 {
+	id -5		# do not change unnecessarily
+	# weight 1.000
+	alg straw
+	hash 0	# rjenkins1
+	item device2 weight 1.000
+}
+rack rack0 {
+	id -3		# do not change unnecessarily
+	# weight 3.000
+	alg straw
+	hash 0	# rjenkins1
+	item host0 weight 1.000
+	item host1 weight 1.000
+	item host2 weight 1.000
+}
+root root {
+	id -4		# do not change unnecessarily
+	# weight 4.000
+	alg straw
+	hash 0	# rjenkins1
+	item rack0 weight 4.000
+}
+
+# rules
+rule data {
+	ruleset 3
+	type replicated
+	min_size 2
+	max_size 2
+	step take root
+	step chooseleaf firstn 0 type rack
+	step emit
+}
+
+# choose_args
+choose_args 1 {
+}
+choose_args 2 {
+  {
+    bucket_id -3
+    ids -20 30 -25
+  }
+}
+choose_args 3 {
+  {
+    bucket_id -3
+    weight_set [
+      [ 1.000 2.000 5.000 ]
+      [ 3.000 2.000 5.000 ]
+    ]
+    ids -20 -30 -25
+  }
+}
+choose_args 4 {
+  {
+    bucket_id -2
+    weight_set [
+      [ 1.000 ]
+      [ 3.000 ]
+    ]
+  }
+}
+choose_args 5 {
+  {
+    bucket_id -1
+    ids -450
+  }
+}
+choose_args 6 {
+  {
+    bucket_id -1
+    ids -450
+  }
+  {
+    bucket_id -2
+    weight_set [
+      [ 1.000 ]
+      [ 3.000 ]
+    ]
+  }
+  {
+    bucket_id -3
+    weight_set [
+      [ 1.000 2.000 5.000 ]
+      [ 3.000 2.000 5.000 ]
+    ]
+    ids -20 -30 -25
```

Comment:
weight_set uses [ ] for the lists.. maybe ids should too?

ids [ -20 -30 -25 ]